HumanExposure/factotum
# 786 Repeated edits fail on QA page

Ticket: #786 

The validation process on the Extracted Text QA page was causing errors and it was preventing users from saving more than one round of changes to children of an extracted text object.
In order to resolve this problem, I made several changes to the Extracted Text QA page.
*The Approval Validation Process*
An Extracted Text object can be approved if:
- none of its child records have been edited
OR
- its child records have been edited AND a QA Note has been added

We were previously enforcing this business rule when we validated the QANotes form. That was not the appropriate place to do it. Now the validation process uses a method on the ExtractedText model. The `is_approvable()` method contains the conditional logic, which can be updated later as the approval workflow changes.

*QA Notes Editing*
Previously, a user could not save QA Notes without clicking "Approve." The notes form was where the system checked to see whether the extractedtext object was eligible for approval. This was linked to the problem that inspired this ticket.
I have replaced the old form with an AJAX approach that allows the user to save QA notes at any time in the process without refreshing the entire page. 

*Approval Endpoint*
The approval process was previously being managed in the `qa_extraction_script()` method in `views/qa.py`, based on a conditional fork. I have moved that logic into a separate URL/view endpoint. 

*Warning and Approval Messages*
Rather than using form validation to alert the user to a failed attempt at approval, I have added Messages to the main template. If the user clicks Approve on an extracted text record with modified child records but no QA notes, an error message appears when the page refreshes.
